pkg/vadmin/fetch_node_state_vc_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+ (c) Copyright [2021-2023] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vadmin
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	vops "github.com/vertica/vcluster/vclusterops"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/fetchnodestate"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// mock version of VFetchNodeState() that is invoked inside VClusterOps.FetchNodeState()
+func (m *MockVClusterOps) VFetchNodeState(options *vops.VFetchNodeStateOptions) ([]vops.NodeInfo, error) {
+	// verify basic options
+	err := m.VerifyBasicOptions(&options.DatabaseOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// verify input hosts
+	if len(options.RawHosts) == 0 {
+		return nil, fmt.Errorf("failed to retrieve hosts")
+	}
+
+	// build a node info list
+	dbName := TestDBName
+	var nodeInfoList []vops.NodeInfo
+	for i := 1; i <= 3; i++ {
+		nodeName := fmt.Sprintf("v_%s_node000%d", dbName, i)
+		nodeInfo := vops.NodeInfo{Name: nodeName, State: "UP"}
+		nodeInfoList = append(nodeInfoList, nodeInfo)
+	}
+	return nodeInfoList, nil
+}
+
+var _ = Describe("fetch_node_state_vc", func() {
+	ctx := context.Background()
+
+	expectedResults := make(map[string]string)
+	for i := 1; i <= 3; i++ {
+		nodeName := fmt.Sprintf("v_%s_node000%d", TestDBName, i)
+		expectedResults[nodeName] = "UP"
+	}
+	initiatorName := "v" + TestDBName + "_node0001"
+
+	It("should call vcluster-ops library with fetch_node_state task", func() {
+		dispatcher := mockVClusterOpsDispatcher()
+		actualResults, ctrlRes, err := dispatcher.FetchNodeState(ctx,
+			fetchnodestate.WithHost(initiatorName, TestInitiatorIP))
```

Comment:
The initiator and the requests hosts are separate arguments. We should have both options given. The hosts being a set of nodes that we think are down and the initiator is the machine we talk to.